src/amalgam/gen/rvv.c

```diff
@@ -201,6 +202,49 @@ void xnn_f32_rminmax_ukernel__rvv_u8v(
   output[1] = __riscv_vfmv_f_s_f32m1_f32(__riscv_vfredmax_vs_f32m8_f32m1(t1, fmax, N));
 }
 
+void xnn_f32_rsum_ukernel__rvv_u4v(
+    size_t batch,
+    const float* input,
+    float* output,
+    const union xnn_f32_scale_params params[restrict XNN_MIN_ELEMENTS(1)])
+{
+  assert(batch != 0);
+  assert(batch % sizeof(float) == 0);
+  assert(input != NULL);
+  assert(output != NULL);
+
+  batch >>= XNN_LOG2_SIZEOF_FLOAT;
+  size_t n = __riscv_vsetvl_e32m1(batch);
+  vfloat32m1_t acc_f32v = __riscv_vfmv_s_f_f32m1(0.f, n);
+  n = __riscv_vsetvl_e32m4(batch);
+  vfloat32m4_t sum0_f32v = __riscv_vfmv_s_f_f32m4(0.f, n);
+  vfloat32m4_t sum1_f32v = __riscv_vfmv_s_f_f32m4(0.f, n);
+  if (batch >= n * 4) {
+    for (; batch >= n * 4; batch -= n * 4) {
```

Comment:
I think `if statement` is unnecessary. The `for-loop statement` has already implied.